app/helpers/course/material/folders_helper.rb

```diff
@@ -8,4 +8,10 @@ module Course::Material::FoldersHelper
   def display_folder(folder)
     folder.root? ? component.settings.title || t('course.material.sidebar_title') : folder.name
   end
+
+  # Involve Course#advance_start_at_duration when deciding if a warning needs to be shown
+  # to staff.
+  def warn_unseen_folder?(folder)
+    (folder.start_at - (folder&.course&.advance_start_at_duration || 0)) > Time.zone.now
```

Comment:
Suggest we shift this part into the model -> `folder.start_at - (folder&.course&.advance_start_at_duration || 0)`, do you think it will be a good idea to expose it as a method instead? If this is useful for lesson plan items we can put it in a concern and include it for lesson plan items too.